// @ismaestro/ngx-example-library/fesm2015/ismaestro-ngx-example-library.js
import { Injectable, Component, Input, APP_INITIALIZER, InjectionToken, NgModule, defineInjectable } from '@angular/core';
import * as moment_ from 'moment';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgxExampleLibraryService {
    constructor() {
    }
}
NgxExampleLibraryService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
NgxExampleLibraryService.ctorParameters = () => [];
/** @nocollapse */ NgxExampleLibraryService.ngInjectableDef = defineInjectable({ factory: function NgxExampleLibraryService_Factory() { return new NgxExampleLibraryService(); }, token: NgxExampleLibraryService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const /** @type {?} */ moment = moment_;
class NgxExampleLibraryComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.date = moment().locale(this.locale).format('dddd, D MMMM, YYYY LT');
    }
}
NgxExampleLibraryComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-ngx-example-library',
                template: `<span>{{date}}</span>`,
                styles: []
            },] },
];
/** @nocollapse */
NgxExampleLibraryComponent.ctorParameters = () => [];
NgxExampleLibraryComponent.propDecorators = {
    "locale": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const /** @type {?} */ OPTIONS = new InjectionToken('OPTIONS');
/**
 * @param {?} options
 * @return {?}
 */
function initialize(options) {
    console.log('Angular library has started with this options: ', options);
    return function () {
    };
}
class NgxExampleLibraryModule {
    /**
     * @param {?=} options
     * @return {?}
     */
    static forRoot(options) {
        return {
            ngModule: NgxExampleLibraryModule,
            providers: [
                { provide: OPTIONS, useValue: options },
                {
                    provide: APP_INITIALIZER,
                    useFactory: initialize,
                    deps: [OPTIONS],
                    multi: true
                },
                NgxExampleLibraryService
            ]
        };
    }
}
NgxExampleLibraryModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [NgxExampleLibraryComponent],
                exports: [NgxExampleLibraryComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { NgxExampleLibraryService, NgxExampleLibraryComponent, OPTIONS, initialize, NgxExampleLibraryModule };
