// @ismaestro/ngx-example-library/esm2015/lib/ngx-example-library.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import * as moment_ from 'moment';
const /** @type {?} */ moment = moment_;
export class NgxExampleLibraryComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.date = moment().locale(this.locale).format('dddd, D MMMM, YYYY LT');
    }
}
NgxExampleLibraryComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-ngx-example-library',
                template: `<span>{{date}}</span>`,
                styles: []
            },] },
];
/** @nocollapse */
NgxExampleLibraryComponent.ctorParameters = () => [];
NgxExampleLibraryComponent.propDecorators = {
    "locale": [{ type: Input },],
};
function NgxExampleLibraryComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxExampleLibraryComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxExampleLibraryComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    NgxExampleLibraryComponent.propDecorators;
    /** @type {?} */
    NgxExampleLibraryComponent.prototype.locale;
    /** @type {?} */
    NgxExampleLibraryComponent.prototype.date;
}
