// @ismaestro/ngx-example-library/esm5/lib/ngx-example-library.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { APP_INITIALIZER, InjectionToken, NgModule } from '@angular/core';
import { NgxExampleLibraryComponent } from './ngx-example-library.component';
import { NgxExampleLibraryService } from './ngx-example-library.service';
export var /** @type {?} */ OPTIONS = new InjectionToken('OPTIONS');
/**
 * @record
 */
export function SampleModuleOptions() { }
function SampleModuleOptions_tsickle_Closure_declarations() {
    /** @type {?} */
    SampleModuleOptions.prototype.config;
}
/**
 * @param {?} options
 * @return {?}
 */
export function initialize(options) {
    console.log('Angular library has started with this options: ', options);
    return function () {
    };
}
var NgxExampleLibraryModule = /** @class */ (function () {
    function NgxExampleLibraryModule() {
    }
    /**
     * @param {?=} options
     * @return {?}
     */
    NgxExampleLibraryModule.forRoot = /**
     * @param {?=} options
     * @return {?}
     */
    function (options) {
        return {
            ngModule: NgxExampleLibraryModule,
            providers: [
                { provide: OPTIONS, useValue: options },
                {
                    provide: APP_INITIALIZER,
                    useFactory: initialize,
                    deps: [OPTIONS],
                    multi: true
                },
                NgxExampleLibraryService
            ]
        };
    };
    NgxExampleLibraryModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [NgxExampleLibraryComponent],
                    exports: [NgxExampleLibraryComponent]
                },] },
    ];
    return NgxExampleLibraryModule;
}());
export { NgxExampleLibraryModule };
function NgxExampleLibraryModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxExampleLibraryModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxExampleLibraryModule.ctorParameters;
}
