// @ismaestro/ngx-example-library/esm5/lib/ngx-example-library.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import * as moment_ from 'moment';
var /** @type {?} */ moment = moment_;
var NgxExampleLibraryComponent = /** @class */ (function () {
    function NgxExampleLibraryComponent() {
    }
    /**
     * @return {?}
     */
    NgxExampleLibraryComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.date = moment().locale(this.locale).format('dddd, D MMMM, YYYY LT');
    };
    NgxExampleLibraryComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-ngx-example-library',
                    template: "<span>{{date}}</span>",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    NgxExampleLibraryComponent.ctorParameters = function () { return []; };
    NgxExampleLibraryComponent.propDecorators = {
        "locale": [{ type: Input },],
    };
    return NgxExampleLibraryComponent;
}());
export { NgxExampleLibraryComponent };
function NgxExampleLibraryComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxExampleLibraryComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxExampleLibraryComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    NgxExampleLibraryComponent.propDecorators;
    /** @type {?} */
    NgxExampleLibraryComponent.prototype.locale;
    /** @type {?} */
    NgxExampleLibraryComponent.prototype.date;
}
